src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -408,47 +395,35 @@ var ReactCompositeComponentMixin = {
     updateQueue
   ) {
     var Component = this._currentElement.type;
-    var instanceOrElement;
+
+    var debugID = 0;
+    if (__DEV__) {
+      debugID = this._debugID;
+    }
+
     if (doConstruct) {
-      if (__DEV__) {
-        if (this._debugID !== 0) {
-          ReactInstrumentation.debugTool.onBeginLifeCycleTimer(
-            this._debugID,
-            'ctor'
-          );
-        }
-      }
-      instanceOrElement = new Component(publicProps, publicContext, updateQueue);
-      if (__DEV__) {
-        if (this._debugID !== 0) {
-          ReactInstrumentation.debugTool.onEndLifeCycleTimer(
-            this._debugID,
-            'ctor'
-          );
-        }
+      if (__DEV__ && debugID !== 0) {
+        return measureLifeCyclePerf(
+          () => new Component(publicProps, publicContext, updateQueue),
+          debugID,
+          'ctor'
```

Comment:
FYI, this was an issue with ES3 but is no longer an issue with ES5 and we do have a transform for it.

<img width="281" alt="screen shot 2016-08-23 at 3 47 25 pm" src="https://cloud.githubusercontent.com/assets/197597/17912502/ecc708f4-6948-11e6-80f8-4969d55461f6.png">